src/util/misc.js

```diff
@@ -221,50 +237,185 @@
+
+          bisector = fabric.util.getBisector(scaledA, scaledB, scaledC);
+        } else {
+          bisector = fabric.util.getBisector(A, B, C);
+        }
+        
+        var bisectorVector = bisector.vector,
             alpha = bisector.angle,
             scalar,
             miterVector;
+
         if (options.strokeLineJoin === 'miter') {
           scalar = -s / Math.sin(alpha / 2);
+
           miterVector = new fabric.Point(
             bisectorVector.x * scalar * strokeUniformScalar.x,
             bisectorVector.y * scalar * strokeUniformScalar.y
           );
-          if (Math.hypot(miterVector.x, miterVector.y) / s <= options.strokeMiterLimit) {
+
+          var strokeMiterLimit;
+          if (options.strokeUniform) {
+            var miterLimitLenght = options.strokeMiterLimit * s,
+                miterLimitVector = new fabric.Point(
+                  bisectorVector.x * miterLimitLenght * strokeUniformScalar.x,
+                  bisectorVector.y * miterLimitLenght * strokeUniformScalar.y
+                );
+            strokeMiterLimit = Math.hypot(miterLimitVector.x, miterLimitVector.y) / s;
+            strokeMiterLimit = Math.sqrt(miterLimitVector.x * miterLimitVector.x + miterLimitVector.y * miterLimitVector.y) / s;
+          } else {
+            strokeMiterLimit = options.strokeMiterLimit;
+          }
+
+          if (Math.sqrt(miterVector.x * miterVector.x + miterVector.y * miterVector.y) / s <= strokeMiterLimit) {
             coords.push(A.add(miterVector));
             coords.push(A.subtract(miterVector));
+            return;
+          } 
+        }
+
+        if (options.strokeLineJoin === 'bevel' || options.strokeLineJoin === 'miter') { // miter greater than stroke miter limit
+
+          var AB = fabric.util.createVector(
+                    options.strokeUniform ? scaledA : A,
+                    options.strokeUniform ? scaledB : B
+                  ),
+              hatOrthogonalAB = fabric.util.getOrthogonalUnitVector(AB),
+              orthogonalAB = new fabric.Point(
+                hatOrthogonalAB.x * s * strokeUniformScalar.x, 
+                hatOrthogonalAB.y * s * strokeUniformScalar.y 
+              );
+
+          var AC = fabric.util.createVector(
+                    options.strokeUniform ? scaledA : A,
+                    options.strokeUniform ? scaledC : C
+                  ),
+              hatOrthogonalAC = fabric.util.getOrthogonalUnitVector(AC),
+              orthogonalAC = new fabric.Point(
+                hatOrthogonalAC.x * s * strokeUniformScalar.x, 
+                hatOrthogonalAC.y * s * strokeUniformScalar.y 
+              );
+          
+
+          coords.push(A.add(orthogonalAB));
+          coords.push(A.subtract(orthogonalAB));
+
+          coords.push(A.add(orthogonalAC));
+          coords.push(A.subtract(orthogonalAC));
+
+          return;
+        }
+
+        if (options.strokeLineJoin === 'round') {
+
+          if (alpha > PiBy2) {
```

Comment:
This seems redundant as well
Seems like it belongs to the previous block

And we should use `else if` for readability if possible